dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/JdbcEnrollmentStore.java

```diff
@@ -0,0 +1,641 @@
+          join note n on n.noteid = en.noteid
+          join userinfo u on u.userinfoid = n.lastupdatedby
+          where en.enrollmentid = e.enrollmentid
+      ) notes on true
+    """);
+  }
+
+  private void addLeftLateralAttributeJoin(StringBuilder sql, EnrollmentQueryParams params) {
+    if (params.isIncludeAttributes()) {
+      sql.append(
+          """
+          left join lateral (
+              select json_agg(json_build_object('uid', tea.uid, 'name', tea.name,
+              'code', tea.code, 'value', teav.value, 'encryptedValue', teav.encryptedvalue,
+              'valueType', tea.valuetype, 'confidential', tea.confidential, 'created', teav.created,
+              'lastUpdated', teav.lastupdated, 'storedBy', teav.storedby)) as jsonattributes
+              from trackedentityattributevalue teav
+              join trackedentityattribute tea ON tea.trackedentityattributeid = teav.trackedentityattributeid
+              where teav.trackedentityid = e.trackedentityid
+          ) attrs on true
+          """);
+    }
+  }
+
+  private void addLastUpdatedConditions(
+      StringBuilder sql,
+      EnrollmentQueryParams enrollmentParams,
+      MapSqlParameterSource sqlParams,
+      SqlHelper hlp) {
+    if (enrollmentParams.hasLastUpdatedDuration()) {
+      sql.append(hlp.whereAnd()).append("e.lastupdated >= :lastupdated");
+      sqlParams.addValue(
+          "lastupdated",
+          new Timestamp(nowMinusDuration(enrollmentParams.getLastUpdatedDuration()).getTime()));
+    } else if (enrollmentParams.hasLastUpdated()) {
+      sql.append(hlp.whereAnd()).append("e.lastupdated >= :lastupdated");
+      sqlParams.addValue("lastupdated", new Timestamp(enrollmentParams.getLastUpdated().getTime()));
+    }
+  }
+
+  private void addOrgUnitConditions(
+      StringBuilder sql,
+      EnrollmentQueryParams params,
+      MapSqlParameterSource sqlParams,
+      SqlHelper hlp) {
+    if (params.hasOrganisationUnits()) {
+      if (params.isOrganisationUnitMode(OrganisationUnitSelectionMode.DESCENDANTS)) {
+        sql.append(hlp.whereAnd())
+            .append(getDescendantsQuery(params.getOrganisationUnits(), sqlParams));
+      } else if (params.isOrganisationUnitMode(OrganisationUnitSelectionMode.CHILDREN)) {
+        sql.append(hlp.whereAnd())
+            .append(getChildrenQuery(params.getOrganisationUnits(), sqlParams));
+      } else {
+        sql.append(hlp.whereAnd()).append("en_ou.uid IN (:orgUnitUids)");
+        sqlParams.addValue("orgUnitUids", getUids(params.getOrganisationUnits()));
+      }
+    }
+  }
+
+  private void addProgramConditions(
+      StringBuilder sql,
+      EnrollmentQueryParams params,
+      MapSqlParameterSource sqlParams,
+      SqlHelper hlp) {
+    sql.append(hlp.whereAnd()).append("p.type = :programType");
+    sqlParams.addValue("programType", ProgramType.WITH_REGISTRATION.name());
+
+    if (params.hasProgram()) {
+      sql.append(hlp.whereAnd()).append("p.uid = :programUid");
```

Comment:
When we will push all the ACL to this store, can we have only the program UID in query params? So we avoid to retrieve the program when mapping from operation params to query params

Reply:
Yes, that would make sense.
I'd just mention the same as above, which is that it'd be good to know if we want to make the program a mandatory parameter in the request or not before we start with the ACL validation.